src/pages/basic_data/app_admin/CustomerInstitutionInformationFromCristin.tsx

```diff
@@ -0,0 +1,94 @@
+import { PageSpinner } from '../../../components/PageSpinner';
+import { Organization } from '../../../types/organization.types';
+import { getUnitTopLevelCode } from '../../../utils/institutions-helpers';
+import { getLanguageString } from '../../../utils/translation-helpers';
+
+interface CustomerInstitutionInformationFromCristinProps {
+  cristinId: string | undefined;
+  customerData: Organization | null;
+  setName: (string: string) => void;
+  displayName: string | undefined;
+}
+
+export const CustomerInstitutionInformationFromCristin = ({
+  cristinId,
+  customerData,
+  setName,
+  displayName,
+}: CustomerInstitutionInformationFromCristinProps) => {
+  const { t } = useTranslation();
+  const [topLevelCristinCode, setTopLevelCristinCode] = useState<string | null>();
+  const customerInformationFromCristinQuery = useQuery({
+    queryKey: ['organization', cristinId, customerData],
+    enabled: !!cristinId && !customerData,
+    queryFn: () => {
+      if (cristinId && !customerData) {
+        return fetchOrganization(cristinId);
+      }
+    },
+  });
+
+  useEffect(() => {
+    setTopLevelCristinCode(getUnitTopLevelCode(cristinId));
+  }, [cristinId]);
+
+  const customerInformation = customerData ?? customerInformationFromCristinQuery.data;
+  const displayNameFromCristin = getLanguageString(customerData?.labels, 'nb');
+  if (displayName !== displayNameFromCristin && displayNameFromCristin.length > 0) {
+    setName(displayNameFromCristin);
+  }
+
+  return (
+    <Grid aria-live="polite" aria-busy={customerInformationFromCristinQuery.isFetching} container spacing={2}>
+      {!customerInformation && customerInformationFromCristinQuery.isFetching ? (
+        <Grid item xs={12}>
+          <PageSpinner aria-label={t('basic_data.institutions.loading_institution_information')} />
+        </Grid>
+      ) : (
+        <>
+          <Grid item xs={12} md={3}>
+            <TextField
+              label={t('basic_data.institutions.institution_norwegian_name')}
+              fullWidth
+              disabled
+              value={displayName ?? ''}
+              variant="filled"
+            />
+          </Grid>
+          <Grid item xs={12} md={3}>
+            <TextField
+              label={t('basic_data.institutions.institution_english_name')}
+              fullWidth
+              disabled
+              value={customerInformation?.labels?.en ?? ''}
+              variant="filled"
+            />
+          </Grid>
+          <Grid item xs={12} md={3}>
+            <TextField
+              disabled
+              fullWidth
+              value={customerInformation?.acronym ?? ''}
+              variant="filled"
+              label={t('basic_data.institutions.short_name')}
+            />
+          </Grid>
+          <Grid item xs={12} md={3}>
+            <TextField
+              disabled
+              fullWidth
+              value={topLevelCristinCode ?? ''}
```

Comment:
`useEffect` er veldig sjelden behov for lenger, og ofte et tegn på at man gjort noe rart designmessig.
Ser ut som vi fjerne både `useEffect` og `useState`, og få samme funksjonalitet, og færre re-renders med noe sånt?
```suggestion
              value={getUnitTopLevelCode(cristinId) ?? ''}
```